### ISSUE TITLE
SignalR hub start invokes done() before ready if jQuery Mobile is referenced

### ISSUE THREAD

**Issue by stimpy77** [SignalR/SignalR#850]
Discovered this last night. If a web page references jQuery 1.8.1/1.8.2 and jQuery Mobile 1.2.0, the behavior of the Deferred object when a SignalR hub is starting is such that the done() callback will be invoked early with the Deferred status marked as resolved, while the connection state is still 0. SignalR will continue to set up the connection, though, and the done() callback will be invoked again. Therefore it seems that a consuming developer's workaround until this issue is resolved is to add: 

 if($.connection.hub.state == 0) return;

.. to the top of the done() callback function before any work is performed. This is not an appropriate long-term solution.

Sample repro:
http://pastie.org/5097059
